### PR TITLE
Add a note about reserved namespace prefix. Resolves #21803

### LINKED ISSUE #21803: Document reserved namespaces
**This is a Feature Request**

**What would you like to be added**
Update https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/ to document that namespaces starting `kube-` are reserved.

**Why is this needed**
According to [API conventions](https://github.com/kubernetes/community/blob/eaddc76fd7d5845665cab136eea16d580852dc47/contributors/devel/sig-architecture/api-conventions.md#namespace-names) all namespaces with names that start with `kube-` are reserved.

**Comments**
/kind feature
/language en
/sig architecture
/good-first-issue

### PR DESCRIPTION
**Pull request by rmohta**
Add a note about reserved namespace prefix. Resolves #21803

### CODE REVIEW

**Comment by sftim** [comment]
Hi @rmohta and thanks for starting your first SIG Docs contribution!

Would you be OK to sign the contributor license agreement - see https://github.com/kubernetes/website/pull/21995#issuecomment-647755269 ?
Once you've signed that we can review your PR.

**Comment by rmohta** [comment]
> Hi @rmohta and thanks for starting your first SIG Docs contribution!
> 
> Would you be OK to sign the contributor license agreement - see [#21995 (comment)](https://github.com/kubernetes/website/pull/21995#issuecomment-647755269) ?
> Once you've signed that we can review your PR.

Hello @sftim , As an individual I have signed the CLA. I'm hoping for this one, to be our company first official contribution. sorry this might take some more time, as we sign the CLA.

**Comment by rmohta** [comment]
I signed it

**Comment by rmohta** [comment]
I signed it

**Comment by rmohta** [comment]
/assign @bradtopol

I had updated the commit message, and force pushed. Can you please help remove the label?

**Comment by sftim** [comment]
Hi @rmohta 

The long form commit message “Resolves #21803” is still invalid. You should put that detail in the _description_ of the pull request, and not in the PR title or in the commit message(s).

If you edit the commit message appropriately, the error label will go away.

**Comment by rmohta** [comment]
Thank You @sftim . It's done now.

**Comment by rmohta** [comment]
Ping @sftim . 
Is there anything pending from my side, before this could be reviewed?

File: content/en/docs/concepts/overview/working-with-objects/namespaces.md [hunk 1]
```diff
@@ -43,6 +43,10 @@ resources within the same namespace.
 Creation and deletion of namespaces are described in the [Admin Guide documentation
 for namespaces](/docs/admin/namespaces).
 
+{{< note >}}
+    Avoid creating namespace with prefix `kube-`, since it is reserved for Kubernetes system namespaces.
```
Comment (sftim) [inline content/en/docs/concepts/overview/working-with-objects/namespaces.md:47]:
```suggestion
Namespaces with names that start with `kube-` are reserved for Kubernetes system use.
```

File: content/en/docs/tasks/administer-cluster/namespaces.md [hunk 2]
```diff
@@ -82,6 +82,10 @@ See the [design doc](https://git.k8s.io/community/contributors/design-proposals/
 
 ## Creating a new namespace
 
+{{< note >}}
+    Avoid creating namespace with prefix `kube-`, since it is reserved for Kubernetes system namespaces.
+{{< /note >}}
```
Comment (sftim) [inline content/en/docs/tasks/administer-cluster/namespaces.md:87]: This could be a caution block instead.

**Pull request merged**